benchmark_test.go

```diff
@@ -0,0 +1,299 @@
+package orderbook
+
+import (
+	"math/rand"
+	"runtime"
+	"strings"
+	"sync"
+	"testing"
+	"time"
+
+	decimal "github.com/geseq/udecimal"
+)
+
+func BenchmarkLatency(b *testing.B) {
+	var totalAddHist, totalCancelHist []float64
+	var mu sync.Mutex
+
+	b.RunParallel(func(pb *testing.PB) {
+		for pb.Next() {
+			addHist, cancelHist := runBenchmarkLatency(b)
+
+			mu.Lock()
+			totalAddHist = append(totalAddHist, addHist...)
+			totalCancelHist = append(totalCancelHist, cancelHist...)
+			mu.Unlock()
+		}
+	})
+
+	printResultsWithPercentiles(b, "Add Order", totalAddHist)
+	printResultsWithPercentiles(b, "Cancel Order", totalCancelHist)
+}
+
+func runBenchmarkLatency(b *testing.B) ([]float64, []float64) {
+	seed := time.Now().UnixNano()
+	duration := 30 * time.Second
+	lowerBound := decimal.MustParse("50.0")
+	upperBound := decimal.MustParse("100.0")
+	minSpread := decimal.MustParse("0.25")
+	sched := true
```

Comment:
```suggestion
	sched := false
```

This is useful for a targeted benchmark that purely analyzes processing latencies without scheduler cost but it’s probably not very useful as a general comparison.

Reply:
fixed